### DISCUSSION TITLE
Output of one query should be an input to the other query.

### DISCUSSION THREAD

**Discussion by VSAmrutha** [hasura/graphql-engine#6094] [General]
I wanted to check if there is a way where we can dynamically add output of one query as an input to the other query in Hasura-Graphql.

Lets say there are two unrelated tables(No PK and FK relation), the only thing common in two tables (Table1 and Table2) are that they have two columns i,e "username" and "email".

Now, lets say I have written as query for the table one as follows:
```
query firstQuery {
  table1 {
   username
   email
   table1_id
  }
} 
```
Then we have query for the second table as follows:
```
query secondQuery {
  table2 (where: {username: {_eq: "john"}, email: {_eq: "john@gmail.com"}}) {   
    username
    email
    bookingDetails_id
    table2_id
  }
}
```
Now the requirement is instead of two queries we need one query as follows where the "username" and "email" are input to the second query.
```
query masterQuery {
  table1 {
    username
    email
    table1_id
  }
  table2 (where: {username: {_eq: username}, email: {_eq: email}}) {   
    username
    email
    bookingDetails_id
    table2_id
  }
} 
```
Now why we need to go from table1 query is that, table1 is the connection from the previous set of tables to the next set of tables.

Same way table2 is the connection to the next set of tables, like "bookingDetails_id".

**Comment by SusanAdey** [comment]
Why not define a relationship between the two tables in your Hasura schema on the username and email columns and then use a nested object type query to retrieve the results. The following link shows how this works
 https://hasura.io/docs/1.0/graphql/core/queries/nested-object-queries.html

**Reply by VSAmrutha** [reply]
@SusanAdey  I asked  this question because as there is no relation between  tables. Please let me know if there is a way as mentioned for above query.

**Reply by AsmaRahimAliJafri** [reply]
you can do so by creating a view
